data/src/main/java/com/harman/roomdbapp/repository/RandomNumberRepositoryImp.kt

```diff
@@ -0,0 +1,26 @@
+package com.harman.roomdbapp.repository
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import com.harman.roomdbapp.convertToEntity
+import com.harman.roomdbapp.convertToRNumber
+import com.harman.roomdbapp.db.RandomNumberDataBase
+import com.harman.roomdbapp.domain.model.RandomNumber
+import com.harman.roomdbapp.domain.repository.RandomNumberRepository
+
+class RandomNumberRepositoryImp(private val db: RandomNumberDataBase): RandomNumberRepository {
+    override suspend fun addNumber(number: RandomNumber) {
+        db.getDao().addNumber(number.convertToEntity())
+    }
+
+    override suspend fun getNumbers(): LiveData<List<RandomNumber>> {
```

Comment:
as it returns observable-like object, there is no need to mark it with "suspend", please remove "suspend" here. Also, LiveData is observed in the main thread, but this method can also be observed in the background, in this case we can use `Flow`. Please replace here LiveData to Flow. Useful links: [https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-flow/](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-flow/), [https://developer.android.com/kotlin/flow](https://developer.android.com/kotlin/flow)